lib/bundle/checker.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Bundle
+  module Checker
+    module_function
+
+    NO_ACTION = [].freeze
```

Comment:
Not sure I understand why this is an empty array rather than e.g. `nil` or `false`?

Reply:
It's convenient for the return of all of the `find_actionable` methods to be a list, that way the list can convey what was there or that there was nothing using the same type, enabling use cases for both the early exit and the full check. Full check benefits more from having the check result be an array since it won't stop on the first missing package, which could instead be represented as a simple string.

Having this be written out enables a more readable

```ruby
def exit_early_check(formulae)
     work_to_be_done = …
     if work_to_be_done
        Bundle::Checker.action_required_for(last_checked)
      else
        Bundle::Checker::NO_ACTION
      end
end
```

which would otherwise be a rather confusing

```ruby
def exit_early_check(formulae)
     work_to_be_done = …
     if work_to_be_done
        [last_checked]
      else
        []
      end
end
```

which is what I had before my brain said, "I can't remember the significance of that".